string2pdb/stringscore

```diff
@@ -20,6 +20,70 @@ import requests
 import argparse
 import pandas as pd
 from io import StringIO
+import re
+
+# Load the aliases file:
+# I assume the file must be moved to /data/databases/ 
+aliases_df = pd.read_csv('/data/user/elenikia/9606.protein.aliases.v12.0.txt', sep='\t', header=None, names=['STRING_ID', 'Alias', 'Source'])
+
+def string2uniprot(string_id):
+
+    # Regex to check if the format matches a reviewed Uniprot entry:
+    reviewed_pattern = re.compile(r'^[A-Z][0-9][A-Z0-9]{4}$')  
+   
+    # Filter rows for STRING ID and 'Ensembl_HGNC_uniprot_ids' source:
+    ids_rows = aliases_df[(aliases_df['STRING_ID'] == string_id) & 
+                      (aliases_df['Source'] == 'Ensembl_HGNC_uniprot_ids')]
+
+    # If no results, check for Uniprot_AC alternatively:
+    if ids_rows.empty:
+        AC_rows = aliases_df[(aliases_df['STRING_ID'] == string_id) & 
+                      (aliases_df['Source'] == 'UniProt_AC')]
+
+        uniprot_acs = AC_rows['Alias'].tolist()
+
+        if not uniprot_acs:
+            print(f"Error: No Uniprot AC found for String ID {string_id}. Exiting")
+            exit(1)
+
+        # Filter UniProt ACs that match the reviewed pattern:
+        reviewed_acs = [ac for ac in uniprot_acs if reviewed_pattern.match(ac)]
+
+        if not reviewed_acs:
+            print(f"Error: No valid reviewed UniProt AC found for STRING ID {string_id}")
+            exit(1)
+
+        if len(reviewed_acs) > 1:
+            # Here (unlike what I do when reviewed_ids >= 1), I return only the most recent one, 
+            # because the rest are either unreviewed, or older versions.
+            # Return the most recent one:
+            reviewed_acs.sort()
```

Comment:
are we sure that this gives us the "most recent" AC? I couldn't find if ACs are assigned chronologically

Reply:
You are absolutely right about this, there is no chronological order. I digged into the cases that do not have Ensembl_HGNC_uniprot_ids alias, and they might have 1 or more Uniprot_AC. In those cases, the id that is returned by mentha, is the one that is present in the source: UniProt_DR_neXtProt. This row is exactly one for all the String IDs. I could directly extract the Uniprot AC from there for all cases but in this way I would not account for examples that have multiple Ensembl_HGNC_uniprot_ids, for which mentha includes all of them. That's why in summary, I am parsing the Ensembl_HGNC_uniprot_ids (and retrieve as many as I find), and only in case there is none, I extract the unique UniProt_DR_neXtProt.

Reply:
Here are some examples of cases with no Ensembl_HGNC_uniprot_ids available, that are present in mentha: Q8IYH5
Q9NQM4 Q3B726 Q9ULR0 Q9H334 Q8IYA8 P15309 Q96NB1. These ids are always found in UniProt_DR_neXtProt row in the aliases file